### ISSUE TITLE
[Bug] Format error when using `dataloader=None` in `train_grad.py` 

### ISSUE THREAD

**Issue by inafergra** [pasqal-io/qadence#503]
Using `dataloader=None` leads to a format error when calling `print_metrics()` in `train()` to print the loss value of the last iteration of the training process, because a `torch.Tensor` is passed instead of a `float`. 

It should be solved by calling `loss = loss.item()` in this case.
